[LuBolin] iP

LuBolin opened the pull request:
# InnKeeper
> _Welcome to my inn 🍺!_

The inn keeper is here to help you keep track of your tasks.
He is
- text-based
- easy to learn
- a chill guy

All you need to do is,
1. download it from [here](https://github.com/LuBolin/ip/releases/).
2. double-click it.
3. add your tasks.
4. let the inn keeper manage your tasks for you 👍.

**Cheers!**

Features:
- [x] Ascii InnKeeper greeting
- [ ] Managing tasks
- [ ] Reminders (coming soon, depending on demand)

Here is the `run()` function in InnKeeper.java, just for fun:

```java
public void run(){
    try{
        storage.readTasksFromFile(tasks, storage, ui);
    } catch (IOException e) {
        ui.printMessage("Error when loading saved tasks: " + e.getMessage());
    }
    ui.setInitialized();
    
    ui.printGreetings();

    boolean isExit = false;
    while (!isExit){
        try {
            String fullCommand = ui.readCommand();
            ui.printLine(); // show the divider line ("_______")
            Command c = inputParser.parseUserInput(fullCommand);
            TerminationType termination = c.execute(tasks, storage, ui);
            isExit = (termination == TerminationType.TERMINATE);
        } catch (Exception e) {
            ui.printMessage(e.getMessage());
        } finally {
            ui.printLine();
        }
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

LuBolin committed:
Renamed chat bot from Duke to InnKeeper.
Renamed main class accordingly.

LuBolin committed:
Added initial skeleton, which contains only greetings.

LuBolin committed:
Cleaned up greetings by grouping into functions.

LuBolin committed:
Fixed grammar, added line separator before farewell (which does not fit level 0's example output, but should work better for future levels).

LuBolin committed:
Added support for echoing of messages. "bye" to exit.
Updated ChatBot's dialogue for more personality.

LuBolin committed:
Added support for list.

LuBolin committed:
Added support for marking done / not done.

LuBolin committed:
Added support for event types (todo, event, deadline).

LuBolin committed:
Cleaned up commented code. Added closed input stream / eof handling.

LuBolin committed:
Added telling of how many items in the list when adding.

LuBolin committed:
Cleaned up println(), it now consistently print 1 empty line after each command.

LuBolin committed:
Modified runtest.bat for semi-automated testing,

LuBolin committed:
Modified some wordings for error handling.
Error handling been implemented previously, without me knowing that it is a later on "increment".

LuBolin committed:
Added support for the delete command.

LuBolin committed:
Add Innkeeper package

LuBolin committed:
Add saving function

LuBolin committed:
Clean up readfile to not print to console.

LuBolin committed:
Add placeholder save file

LuBolin committed:
Merge branch 'branch-Level-7'

LuBolin committed:
Add support for parsing datets as yyyy-mm-dd

LuBolin committed:
Add hint for date formatting

LuBolin committed:
Merge branch 'branch-Level-8'

LuBolin committed:
Restructured OOP.

LuBolin committed:
Merge branch 'branch-More-OOP'

LuBolin committed:
Merge branch 'add-gradle-support'

LuBolin committed:
Add Gradle, update manual test

LuBolin committed:
Add JUnit test for 2 classes

LuBolin committed:
Update gradle to match build

LuBolin committed:
Add javadoc for root classes

LuBolin committed:
Add JavaDoc for tasks

LuBolin committed:
Add JavaDoc for commands

LuBolin committed:
Satisfy coding standards

LuBolin committed:
Add support for find command

LuBolin committed:
Merge branch 'branch-A-JavaDoc'

LuBolin committed:
Merge branch 'branch-A-CodingStandard'

LuBolin committed:
Merge branch 'branch-A-JavaDoc' into Level-9

# Conflicts:
#	data/tasks.txt

LuBolin committed:
Remove I/O redirection style auto-testing

LuBolin committed:
Switch to JavaFX

LuBolin committed:
Merge branch 'branch-Level-10'

jiangsuwangjing commented on src/main/java/innkeeper/Storage.java:
```diff
@@ -0,0 +1,124 @@
+                    System.out.println("Failed to write to file " + FILE_PATH);
+                    System.out.println("File not found and could not creat it.");
+                    return;
+                }
```
It might be better to create an Exception for this case and handle it.

jiangsuwangjing commented on src/main/java/innkeeper/Storage.java:
```diff
@@ -0,0 +1,124 @@
+                if (newCommand instanceof TodoCommand
+                        || newCommand instanceof innkeeper.command.DeadlineCommand
+                        || newCommand instanceof innkeeper.command.EventCommand) {
+                    try{
```
This nested try-catch block may not be necessary.

jiangsuwangjing commented on src/main/java/innkeeper/TaskList.java:
```diff
@@ -0,0 +1,84 @@
+            String message = "My notebook is full! I can't add any more notes.\n" +
+                    "My old brain can only remember up to " + MAX_LIST_SIZE + " tasks.";
+            throw new ListFullException(message);
+        }
```
I like this exception, it can be useful for larger projects.

jiangsuwangjing commented on src/test/java/DeadlineCommandTest.java:
```diff
@@ -0,0 +1,30 @@
+
+public class DeadlineCommandTest {
+    @Test
+    public void parameterParseTest(){
```
The test case name can be more detailed, showing the condition and expected outcome.

jiangsuwangjing commented on src/main/java/innkeeper/InnKeeper.java:
```diff
@@ -0,0 +1,51 @@
+        }
+    }
+
+    public String getGreetings(boolean withAsciiArt) {
+        return ui.getGreetings(withAsciiArt);
+    }
```
This function seems trivial and could be handled inside functions or by UI itself.

jiangsuwangjing reviewed (commented):
Overall good adherence to coding quality and OOP standards.

LuBolin committed:
Add CheckStyle

Add CheckStyle to aid code quality.

LuBolin committed:
Merge branch 'branch-A-CheckStyle'

LuBolin committed:
Add assertion for assumptions

Assert for existence of file when writing to save the list.
Assert for user input string to not be a null string.

LuBolin committed:
Fix code quality

Made fixes based on CheckStyle.

LuBolin committed:
Merge pull request #1 from LuBolin/branch-A-Assertions

Add assertion for assumptions

LuBolin committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/innkeeper/InputParser.java

LuBolin committed:
Merge pull request #2 from LuBolin/branch-A-CodeQuality

Merge code quality branch to main

LuBolin committed:
Add fuzzy find

Add fuzzy find command to match a % of keyword.

LuBolin committed:
Merge pull request #3 from LuBolin/branch-BCD-Extension

Add fuzzy find

LuBolin committed:
Fix style issued suggested by SE bot

LuBolin committed:
Add Ui.png, add user guide into README

LuBolin committed:
Add comment about Unsupported JavaFX configuration

LuBolin committed:
Add instructions about exiting innkeeper

LuBolin committed:
Update Ui.png to have a slight border
